eslint-rules/warning-and-invariant-args.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright 2015, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @emails react-core
+*/
+
+'use strict';
+
+/**
+ * The warning() and invariant() functions take format strings as their second
+ * argument.
+ */
+
+module.exports = function(context) {
+  // we also allow literal strings and concatinated literal strings
+  function getLiteralString(node) {
+    if (node.type === 'Literal' && typeof node.value === 'string') {
+      return node.value;
+    } else if (node.type === 'BinaryExpression' && node.operator === '+') {
+      var l = getLiteralString(node.left);
+      var r = getLiteralString(node.right);
+      if (l !== null && r !== null) {
+        return l + r;
+      }
+    }
+    return null;
+  }
+
+  return {
+    CallExpression: function(node) {
+      // This could be a little smarter by checking context.getScope() to see
+      // how warning/invariant was defined.
+      var isWarningOrInvariant =
+        node.callee.type === 'Identifier' &&
+        (node.callee.name === 'warning' || node.callee.name === 'invariant');
+      if (isWarningOrInvariant) {
```

Comment:
Let's return early if it's not warning or invariant instead of nesting the rest of the code in an if block.